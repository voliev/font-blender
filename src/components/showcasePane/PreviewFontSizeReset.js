import React from "react";
import PropTypes from "prop-types";
import "./styles/PreviewFontSizeReset.css";

const PreviewFontSizeReset = ({ resetFontSizeHandler }) => (
  <button
    className="PreviewFontSizeReset btn-blue-rect"
    onClick={resetFontSizeHandler}
    value="reset"
    title="Reset font size"
  >
    Reset
  </button>
);

PreviewFontSizeReset.propTypes = {
  resetFontSizeHandler: PropTypes.func.isRequired
};

export default PreviewFontSizeReset;
