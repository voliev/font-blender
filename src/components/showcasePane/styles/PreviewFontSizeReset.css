.PreviewFontSizeReset {
  height: 2rem;
  padding: var(--space-xs) var(--space-md);
}
