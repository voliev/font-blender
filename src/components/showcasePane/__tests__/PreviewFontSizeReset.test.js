import React from "react";
import { mount, shallow } from "enzyme";
import PreviewFontSizeReset from "../PreviewFontSizeReset";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    resetFontSizeHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<PreviewFontSizeReset {...props} />)
  };
};

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { resetFontSizeHandler, wrapper } = setup(mount);

  expect(typeof wrapper.prop("resetFontSizeHandler")).toBe("function");
  expect(wrapper.prop("resetFontSizeHandler")).toBe(resetFontSizeHandler);
});

it("should handle button click", () => {
  const { resetFontSizeHandler, wrapper } = setup(mount);

  wrapper.find("button").simulate("click");

  expect(resetFontSizeHandler).toHaveBeenCalledTimes(1);
});
