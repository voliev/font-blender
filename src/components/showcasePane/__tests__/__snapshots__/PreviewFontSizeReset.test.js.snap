// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<button
  className="PreviewFontSizeReset btn-blue-rect"
  onClick={[MockFunction]}
  title="Reset font size"
  value="reset"
>
  Reset
</button>
`;
